packages/dds/tree/src/feature-libraries/modular-schema/modularChangeTypes.ts

```diff
@@ -31,12 +34,36 @@ export interface ModularChangeset extends HasFieldChanges {
 	readonly revisions?: readonly RevisionInfo[];
 	readonly fieldChanges: FieldChangeMap;
 	readonly nodeChanges: ChangeAtomIdMap<NodeChangeset>;
+
+	// TODO: Should this be merged with `nodeChanges`?
+	readonly nodeToParent: ChangeAtomIdMap<FieldId>;
+	readonly nodeAliases: ChangeAtomIdMap<NodeId>;
+	readonly crossFieldKeys: CrossFieldKeyTable;
 	readonly constraintViolationCount?: number;
 	readonly builds?: ChangeAtomIdMap<TreeChunk>;
 	readonly destroys?: ChangeAtomIdMap<number>;
 	readonly refreshers?: ChangeAtomIdMap<TreeChunk>;
 }
 
+export type CrossFieldKeyTable = TupleBTree<CrossFieldKeyRange, FieldId>;
+export type CrossFieldKeyRange = [
+	CrossFieldTarget,
+	RevisionTag | undefined,
+	ChangesetLocalId,
+	/**
+	 * The length of this range.
+	 * TODO: This does not need to be part of the key and could be part of the value instead.
+	 */
+	number,
+];
+
+export type CrossFieldKey = [CrossFieldTarget, RevisionTag | undefined, ChangesetLocalId];
```

Comment:
Would be nice to make the entries of `CrossFieldKey` readonly of that's practical.